dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -5375,11 +5387,11 @@ Sedp::write_participant_message_data(
   if (!(spdp_.available_builtin_endpoints() & BUILTIN_ENDPOINT_PARTICIPANT_MESSAGE_DATA_WRITER)) {
     return DDS::RETCODE_PRECONDITION_NOT_MET;
   }
-
   DDS::ReturnCode_t result = DDS::RETCODE_OK;
+  ParticipantMessageData pmd;
```

Comment:
Initialize pmd.participantGuid (see line 5395).
You may want to add a comment like "Update liveliness for local subscriptions."